### ISSUE TITLE
Consider moving configuration to build time instead of runtime

### ISSUE THREAD

**Issue by emmanuelbernard** [spinscale/quarkus-logging-ecs#2]
The configuration is update between each run https://github.com/spinscale/quarkus-logging-ecs/blob/master/runtime/src/main/java/de/spinscale/quarkus/logging/ecs/runtime/EcsLoggingConfig.java#L26

The Quarkus philisophy is to fix as many properties as possible to build time to then prepare the frameworks at build time as much as possible. I'd recommend you review the properties and decide which one could be fixed at build time and which one has to be overridable between runs.

**Comment by spinscale** [comment]
thx a ton for all your input, will work on those items as soon as I can.

Regarding this one, I saw that none of the logging extensions in quarkus have set anything but `RUN_TIME` - I suppose that all of those settings in the other implementations (regular json logging or sentry) are also supposed to be changed at runtime via configuration or am I misunderstanding a quarkus feature here?

**Comment by emmanuelbernard** [comment]
ah, they might:
- have been lazy
- realized that fixing any of the properties at build time would lead to bad UX
- realized that fixing any of the properties at build time would not lead to optimizations
    - neither pre initialization of the framework
    - nor ease the dead code elimination in native image

I'd guess lazy but maybe @gsmet knows more

**Comment by gsmet** [comment]
Yeah, I must admit I have a different approach here as I think in the case of logging, it makes a lot of sense to be able to tune the behavior at runtime.

So if moving them to build time does not bring any improvement in terms of startup time, I would keep the flexibility.
